src/app/pages/account/groups/group-form/group-form.component.ts

```diff
@@ -37,8 +42,76 @@ export class GroupFormComponent {
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+      { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
+    ]),
+    pageOffset: 18,
+    filter: (options$: Observable<Option[]>, value: string): void => {
+      if (value) {
+        this.normalLoading = true;
+        options$.pipe(untilDestroyed(this)).subscribe((syncOptions) => {
+          this.normalProvider.options$ = of(syncOptions.filter((option: Option) => {
+            this.normalLoading = false;
+            return option.label.toLowerCase().includes(value.toLowerCase())
+                || option.value.toString().toLowerCase().includes(value.toLowerCase());
+          }));
+        });
+      }
+    },
+    onScrollEnd: () => {
+      this.normalProvider.pageOffset += 12;
+      this.normalLoading = true;
+
+      this.normalProvider.options$ = forkJoin([
+        this.normalProvider.options$,
+        of([
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+          { label: 'Rehan - ' + UUID.UUID().substring(0, 6), value: UUID.UUID() },
+        ] as Option[]),
+      ]).pipe(delay(3000)).pipe(map(([arr1, arr2]) => {
+        this.normalLoading = false;
+        return [...arr1, ...arr2];
+      }));
+      this.normalProvider = _.cloneDeep(this.normalProvider);
+      this.cdr.markForCheck();
+    },
+  };
+
```

Comment:
It's nice to see an example of a more advanced provider with static values. I would also like to see an example of simplified version for developer use, something like:
```
normalProvider = new ArrayProvider([
       { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
       { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
       { label: UUID.UUID().substring(0, 6), value: UUID.UUID() },
]);

```

Reply:
Still would like to see this example.